lib/core/utils/preload-cssom.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * Returns a then(able) queue of CSSStyleSheet(s)
+ * @param {Object} ownerDocument document object to be inspected for stylesheets
+ * @param {number} timeout on network request for stylesheet that need to be externally fetched
+ * @param {Function} getSheetFromTextFn a utility function to generate a style sheet from text
+ * @return {Object} queue
+ * @private
+ */
+function loadCssom(ownerDocument, timeout, getSheetFromTextFn) {
+	const q = axe.utils.queue();
+
+	Array.from(ownerDocument.styleSheets).forEach(sheet => {
+		if (sheet.disabled) {
+			return;
+		}
+
+		try {
+			// cannot use if/else here as the catch block won't be triggered (if passes for both relative and external sheets)
```

Comment:
I do not understand this. Why would accessing the `.cssRules` property cause an exception to be thrown? Unless `sheet` is `null` or `undefined`, the JavaScript runtime will allow you to read properties from it:

```js
sheet = 'sheet'
// => "sheet"
sheet.cssRules
// => undefined

sheet = true
// => true
sheet.cssRules
// => undefined

sheet = {}
// => {}
sheet.cssRules
// => undefined

sheet = null
// => null
sheet.cssRules
// => VM352:1 Uncaught TypeError: Cannot read property 'cssRules' of null
    at <anonymous>:1:7
(anonymous) @ VM352:1
```

We're also sure that `sheet` is non-null, since we access `.href` from it on L25.

Reply:
@stephenmathieson - try it.

Spin up a html page with one relative stylesheet, one cross-domain stylesheet and check `document.styleSheets[n].cssRules` on the external stylesheet.

Happy to find an alternative.